tests/phpunit/tests/rest-api/wpApplicationPasswords/createNewApplicationPassword.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * Unit tests covering WP_Application_Passwords::create_new_application_password functionality.
+ *
+ * @package    WordPress
+ * @subpackage REST API
+ */
+
+/**
+ * @covers WP_Application_Passwords::create_new_application_password
+ *
+ * @group  restapi
+ * @group  app_password
+ */
+class Test_WPApplicationPasswords_CreateNewApplicationPassword extends WP_UnitTestCase {
+
+	/**
+	 * Administrator user id.
+	 *
+	 * @var int
+	 */
+	private static $user_id;
+
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		self::$user_id = $factory->user->create(
+			array(
+				'role' => 'administrator',
+			)
+		);
+
+		if ( is_multisite() ) {
+			grant_super_admin( self::$user_id );
+		}
+	}
+
+	public static function wpTearDownAfterClass() {
+		self::delete_user( self::$user_id );
+	}
+
+	/**
+	 * @ticket       51941
+	 * @dataProvider data_returns_wp_error
+	 */
+	public function test_returns_wp_error( $expected, array $args = array(), array $names = array() ) {
```

Comment:
```suggestion
	public function test_create_validation( $expected, array $args = array(), array $names = array() ) {
```

Reply:
Applied to test method and data provider in commit https://github.com/WordPress/wordpress-develop/pull/898/commits/782a8449ff6fac14019c48b08a4564038ad983df.